spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/autoconfig/TraceAutoConfiguration.java

```diff
@@ -283,4 +282,40 @@ public String toString() {
 
 	}
 
+	@Configuration
+	@ConditionalOnMissingClass("io.micrometer.core.instrument.MeterRegistry")
+	static class TraceMetricsNoOpConfiguration {
```

Comment:
should this be

```suggestion
	static class TraceMetricsInMemoryConfiguration {
```

if no-op is desired, that would be different and cheaper.